internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -1,32 +1,233 @@
+			process.FindMainProcessReturns(1234, nil)
+			process.ReadFileReturns([]byte("child1\nchild2"), nil)
+			process.KillReturns(fmt.Errorf("failed to send kill signal"))
+
+			err := manager.Reload(context.Background(), 1)
+
+			Expect(err).To(MatchError("failed to send the HUP signal to NGINX main: failed to send kill signal"))
+			Expect(metrics.IncReloadErrorsCallCount()).To(Equal(1))
+			Expect(verifyClient.WaitForCorrectVersionCallCount()).To(Equal(0))
+		})
+
+		It("times out waiting for correct version", func() {
+			process.FindMainProcessReturns(1234, nil)
+			process.ReadFileReturns([]byte("child1\nchild2"), nil)
+			process.KillReturns(nil)
+			verifyClient.WaitForCorrectVersionReturns(fmt.Errorf("timeout waiting for correct version"))
+
+			err := manager.Reload(context.Background(), 1)
+
+			Expect(err).To(MatchError("timeout waiting for correct version"))
+			Expect(metrics.IncReloadErrorsCallCount()).To(Equal(1))
+		})
+
+		When("MetricsCollector is nil", func() {
+			It("panics", func() {
+				metrics = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					err = manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		When("VerifyClient is nil", func() {
+			It("panics", func() {
+				metrics = &runtimefakes.FakeMetricsCollector{}
+				verifyClient = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					err = manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+	})
+
+	When("running NGINX plus", func() {
+		BeforeEach(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, nil, zap.New(), nil, nil)
+		})
+
+		It("successfully updates HTTP server upstream", func() {
+			Expect(manager.UpdateHTTPServers("test", upstreamServers)).To(Succeed())
+		})
+
+		It("returns no upstreams from NGINX Plus API when upstreams are nil", func() {
+			upstreams, err := manager.GetUpstreams()
+
+			Expect(err).To(HaveOccurred())
+			Expect(upstreams).To(BeEmpty())
+		})
+
+		It("successfully returns server upstreams", func() {
+			upstreams := ngxclient.Upstreams{
+				"upstream1": {
+					Zone: "zone1",
+					Peers: []ngxclient.Peer{
+						{ID: 1, Name: "peer1-name"},
+					},
+					Queue:      ngxclient.Queue{Size: 10},
+					Keepalives: 5,
+					Zombies:    2,
+				},
+				"upstream2": {
+					Zone: "zone2",
+					Peers: []ngxclient.Peer{
+						{ID: 2, Name: "peer2-name"},
+					},
+					Queue:      ngxclient.Queue{Size: 20},
+					Keepalives: 3,
+					Zombies:    1,
+				},
+			}
+
+			ngxPlusClient.GetUpstreamsReturns(&upstreams, nil)
+
+			upstreams, err := manager.GetUpstreams()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(upstreams).To(Equal(upstreams))
```

Comment:
This assertion is a little off. We need to store the actual upstreams, returned by `manager.GetUpstreams()` in a variable different from the upstreams we compare them to. 

I recommend renaming upstreams on line 161 to `expUpstreams`